python/hsml/engine/model_engine.py

```diff
@@ -270,10 +270,17 @@ def _set_model_version(
         # Set model version if not defined
         if model_instance._version is None:
             current_highest_version = 0
-            for item in self._dataset_api.list(dataset_model_path, sort_by="NAME:desc")[
-                "items"
-            ]:
+            files = []
+            offset = 0
+            limit = 1000
+            items = self._dataset_api.list(dataset_model_path, sort_by="NAME:desc", offset=offset, limit=limit)["items"]
```

Comment:
If there are no files, is ["items"] still in the returned object or will it fail with an error hard to understand by users? (e.g., no "items" key found)

Reply:
It still returns items, but the value is []